contracts/stabilizer/PegKeeperRegulator.vy

```diff
@@ -232,10 +241,16 @@ def add_peg_keepers(_peg_keepers: DynArray[PegKeeper, MAX_LEN]):
     i: uint256 = len(self.peg_keepers)
     for pk in _peg_keepers:
         assert self.peg_keeper_i[pk] == empty(uint256)  # dev: duplicate
+        pool: StableSwap = pk.pool()
+        success: bool = raw_call(
+            pool.address, _abi_encode(convert(0, uint256), method_id=method_id("price_oracle(uint256)")),
+            revert_on_failure=False
```

Comment:
If you'd like to be definitively sure if a pool is stableswap-ng, you can check if the pool is from a stableswap-ng factory (which is probably cheaper to calc), something like:

```vyper
@internal
@view
def _pool_is_ng(_pool: address) -> bool:

    # Check if pool is from stableswap ng factory:
    return raw_call(
        stableswap_ng_factory,
        concat(
            method_id('get_n_coins(address)'),
            convert(_pool, bytes32),
        ),
        revert_on_failure=False,
        is_static_call=True,
    )
```

Reply:
Are all `price_oracle(uint256)` versions are ng pools? And if there is yet another custom version is it possible to include in ng factory?
Updating regulator leads to updating all peg keepers and controllers, a big vote.

Reply:
i guess you cannot add custom versions, but if you want to be sure if a stableswap pool is from the ng factory, you'd need to check if it was in the factory (by calling factory.get_n_coins(_pool), which reverts if factory's pooldata map doesnt have the pool in it.)

i think what you do is fine of course. but for correctness i would recommend checking if a pool is actually from stableswap-ng factory. up to you!